File: app/src/main/java/com/cn29/aac/datasource/feedentry/db/migration/Migration2_3.java
package com.cn29.aac.datasource.feedentry.db.migration;

import androidx.room.migration.Migration;
import androidx.sqlite.db.SupportSQLiteDatabase;

/**
 * Created by Charles Ng on 12/9/2017.
 */

public class Migration2_3 extends Migration {

  /**
   * Creates a new migration between {@code startVersion} and {@code endVersion}.
   *
   * @param startVersion The start version of the database.
   * @param endVersion The end version of the database after this migration is applied.
   */
  public Migration2_3(int startVersion, int endVersion) {
    super(startVersion, endVersion);
  }

  public static Migration2_3 newInstance() {
    return new Migration2_3(2, 3);
  }

  @Override
  public void migrate(SupportSQLiteDatabase database) {
    database.execSQL("ALTER TABLE feedEntrys ADD COLUMN favourite INTEGER DEFAULT 0 ");
  }
}

File: app/src/main/java/com/cn29/aac/datasource/feedentry/db/migration/Migration2_3.kt
package com.cn29.aac.datasource.feedentry.db.migration

import androidx.room.migration.Migration
import androidx.sqlite.db.SupportSQLiteDatabase

class Migration2_3(startVersion: Int,
                   endVersion: Int) : Migration(startVersion, endVersion) {
    override fun migrate(database: SupportSQLiteDatabase) {
        database.execSQL("ALTER TABLE feedEntrys ADD COLUMN favourite INTEGER NOT NULL DEFAULT 0 ")
    }

    companion object {
        @JvmStatic
        fun newInstance(): Migration2_3 {
            return Migration2_3(2, 3)
        }
    }
}